crates/dojo-language-server/src/bin/language_server.rs

```diff
@@ -0,0 +1,70 @@
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use anyhow::Result;
+use cairo_lang_compiler::db::{RootDatabase, RootDatabaseBuilder};
+use cairo_lang_dojo::plugin::DojoPlugin;
+use cairo_lang_filesystem::db::init_dev_corelib;
+use cairo_lang_language_server::{Backend, State};
+use cairo_lang_plugins::get_default_plugins;
+use cairo_lang_semantic::db::SemanticGroup;
+use cairo_lang_semantic::plugin::SemanticPlugin;
+use cairo_lang_starknet::plugin::StarkNetPlugin;
+use tower_lsp::{LspService, Server};
+
+const CORELIB_DIR_NAME: &str = "cairo/corelib";
+
+trait RootDatabaseBuilderDojo {
+    fn build_language_server(
+        &mut self,
+        path: PathBuf,
+        plugins: Vec<Arc<dyn SemanticPlugin>>,
+    ) -> Result<RootDatabase>;
+}
+
+impl RootDatabaseBuilderDojo for RootDatabaseBuilder {
+    fn build_language_server(
+        &mut self,
+        path: PathBuf,
+        plugins: Vec<Arc<dyn SemanticPlugin>>,
+    ) -> Result<RootDatabase> {
+        let mut db = RootDatabase::default();
+        init_dev_corelib(&mut db, path);
+        db.set_semantic_plugins(plugins);
+        Ok(db)
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    #[cfg(feature = "runtime-agnostic")]
+    use tokio_util::compat::{TokioAsyncReadCompatExt, TokioAsyncWriteCompatExt};
+
+    let (stdin, stdout) = (tokio::io::stdin(), tokio::io::stdout());
+    #[cfg(feature = "runtime-agnostic")]
+    let (stdin, stdout) = (stdin.compat(), stdout.compat_write());
+
+    let mut plugins = get_default_plugins();
+    plugins.push(Arc::new(DojoPlugin {}));
+    plugins.push(Arc::new(StarkNetPlugin {}));
+
+    let mut dir = std::env::current_exe()
```

Comment:
well have to find a way to embed the corelib so execution doesn't depend on the directory structure

Reply:
Eventually we will get rid of the Cairo submodule and refer to Cairo global path